File: tests/node/uncaught_exception.js
// Uncaught exception - NO signal expected
// Node.js handles this entirely in JS-land. The process exits with code 1,
// NOT via a signal. crash-tracer should NOT see this as a crash.
// This is a key false-negative test: a real application error that produces
// no signal at all.

console.error("[node/uncaught_exception] Throwing an unhandled Error...");

function deepCall(n) {
    if (n === 0) {
        throw new Error("Unhandled application error: database connection failed");
    }
    return deepCall(n - 1);
}

// Build up a realistic call stack
deepCall(10);

File: tests/java/UncaughtException.java
// Uncaught exception — NO signal expected
// An unhandled Java exception causes System.exit(1). The JVM shuts down
// cleanly via its normal exit path. No signal is delivered.
// crash-tracer should NOT capture this.

public class UncaughtException {
    public static void main(String[] args) {
        System.err.println("[java/uncaught_exception] Throwing an unhandled RuntimeException...");

        deepCall(10);
    }

    static void deepCall(int n) {
        if (n == 0) {
            throw new RuntimeException("Simulated application failure: connection refused");
        }
        deepCall(n - 1);
    }
}
